### PR TITLE
Fix case insensitive matching when dropping MongoDB schema 

### LINKED ISSUE #15716: MongoDB - DROP SCHEMA with mixed case doesn't work
MongoDB - Drop schema with mixed case schema name(`Test_Case_Insensitive`) doesn't drop the schema

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Fixes #15716

## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# MongoDB
* Fix bug when dropping schemas and `mongodb.case-insensitive-name-matching` is enabled.
  Previously, the statement succeeded, but the schema wasn't dropped. ({issue}`15716`)
```

### CODE REVIEW

**Review by hashhar — APPROVED** [review]
LGTM but I assume you'll rebase after merging https://github.com/trinodb/trino/pull/15719?

**Comment by ebyhr** [comment]
Yes, rebased on upstream to resolve conflicts.

**Review by hashhar — APPROVED** [review]

**Pull request merged**
